### PR TITLE
Release notes for wasmtime-11

### PR DESCRIPTION
**Pull request by elliottt**
Update the release notes for the upcoming 11.0.0 release.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -18,16 +18,54 @@ Unreleased.
...
 ### Fixed
 
 * Only export the top-level preview2 module from wasmtime-wasi when the
   `preview2` feature is enabled.
   [#6615](https://github.com/bytecodealliance/wasmtime/pull/6615)
 
+
+### Cranelift changes
+
+* Tail call implementation has begun in cranelift
```
Comment (cfallin) [inline RELEASES.md:34]: s/cranelift/Cranelift/

File: RELEASES.md [hunk 2]
```diff
@@ -18,16 +18,54 @@ Unreleased.
...
+  [#6598](https://github.com/bytecodealliance/wasmtime/pull/6598)
+  [#6599](https://github.com/bytecodealliance/wasmtime/pull/6599)
+  [#6587](https://github.com/bytecodealliance/wasmtime/pull/6587)
+  [#6568](https://github.com/bytecodealliance/wasmtime/pull/6568)
+  [#6515](https://github.com/bytecodealliance/wasmtime/pull/6515)
+
+* Fix `AuthenticatedRet` when stack bytes are popped in the aarch64 backend
+  [#6634](https://github.com/bytecodealliance/wasmtime/pull/6634)
+
+* Lower SIMD requirements to SSE2 on x64
```
Comment (cfallin) [inline RELEASES.md:57]: We could say a bit more about this here -- it's a pretty nice accomplishment and something I'm not aware of any other engine supporting on x86 (Wasm-SIMD itself is on record as having been designed with modern SSE4 in mind, e.g. [here](https://github.com/bytecodealliance/wasmtime/pull/6224#issuecomment-1553306884).

Perhaps something like "Wasmtime/Cranelift on x86-64 can now execute Wasm-SIMD on baseline SSE2, which all x86-64 processors support (as part of the base x86-64 spec). Previously, SSE4.2 extensions were required. This new work allows Wasm with SIMD extensions to execute on processors produced back to 2003."

File: RELEASES.md [hunk 3]
```diff
@@ -18,16 +18,54 @@ Unreleased.
...
+  [#6625](https://github.com/bytecodealliance/wasmtime/pull/6625)
+
+* The `fcvt_low_from_sint` instruction has been removed
+  [#6565](https://github.com/bytecodealliance/wasmtime/pull/6565)
```
Comment (cfallin) [inline RELEASES.md:61]: Can we say what the replacement is (combo of two other instructions)?

**Review by cfallin — APPROVED** [review]
Thanks!

File: RELEASES.md (hunk 2 above)
Comment (elliottt) [reply RELEASES.md]: I like your suggestion!

**Review by alexcrichton — COMMENTED** [review]
Thanks for writing this up! Looks like a relatively quiet release and the only thing I might add is:

* Component `bindgen!` now generates owned types by default instead of based on how they're used. #6648

**Pull request merged**